File: sample-conference-videochat/sample-conference-videochat/Components/Calls/CallViewController/StatsView.m
//
//  StatsView.m
//  sample-videochat-webrtc-old
//
//  Copyright (c) 2017 QuickBlox. All rights reserved.
//

#import "StatsView.h"

static NSString * const kQMStatsReportPlaceholderText = @"Loading stats report...";

@implementation StatsView {
    UILabel *_statsLabel;
}

- (instancetype)initWithFrame:(CGRect)frame {
    
    self = [super initWithFrame:frame];
    if (self != nil) {
        
        _statsLabel = [[UILabel alloc] initWithFrame:frame];
        _statsLabel.text = kQMStatsReportPlaceholderText;
        _statsLabel.numberOfLines = 0;
        _statsLabel.font = [UIFont fontWithName:@"Roboto" size:12];
        _statsLabel.adjustsFontSizeToFitWidth = YES;
        _statsLabel.minimumScaleFactor = 0.6;
        _statsLabel.textColor = [UIColor greenColor];
        [self addSubview:_statsLabel];
        self.backgroundColor = [UIColor colorWithWhite:0 alpha:.6];
    }
    return self;
}

- (void)setStats:(NSString *)stats {
    if (stats == nil) {
        stats = kQMStatsReportPlaceholderText;
    }
    _statsLabel.text = stats;
}

- (void)layoutSubviews {
    _statsLabel.frame = self.bounds;
}

- (void)setHidden:(BOOL)hidden {
    
    if (hidden) {
        
        [self setStats:nil];
    }
    
    [super setHidden:hidden];
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/sample-conference-videochat-swift/Components/Calls/CallViewController/StatsView.swift
//
//  StatsView.swift
//  sample-conference-videochat-swift
//
//  Created by Vladimir Nybozhinsky on 04.10.2018.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

struct StatsViewConstants {
    static let kQMStatsReportPlaceholderText = "Loading stats report..."
}

class StatsView: UIView {
    
    lazy private var statsLabel: UILabel = {
        let statsLabel = UILabel(frame: bounds)
        statsLabel.text = StatsViewConstants.kQMStatsReportPlaceholderText
        statsLabel.numberOfLines = 0
        statsLabel.font = UIFont(name: "Roboto", size: 12)
        statsLabel.adjustsFontSizeToFitWidth = true
        statsLabel.minimumScaleFactor = 0.6
        statsLabel.textColor = UIColor.green
        return statsLabel
    }()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        addSubview(statsLabel)
        backgroundColor = UIColor(white: 0, alpha: 0.6)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func setStats(_ stats: String?) {
        var stats = stats
        if stats == nil {
            stats = StatsViewConstants.kQMStatsReportPlaceholderText
        }
        statsLabel.text = stats
    }
    
    override func layoutSubviews() {
        statsLabel.frame = bounds
    }
    
    override var isHidden: Bool {
        didSet {
            if isHidden == true {
                setStats(nil)
            }
        }
    }
}
